### PR TITLE
Fix command string truncation in bash string cmds

### PR DESCRIPTION
**Pull request by Birdrock**
## Is there a related GitHub Issue?
#1439 

## What is this change about?
Fixes an issue with how command and args were processed. The issue lead to a situation where args would be dropped.

## Does this PR introduce a breaking change?
No.

## Acceptance Steps
Push a ruby rackup app (e.g.: CATS hello-world)

## Tag your pair, your PM, and/or team
@akrishna90 @davewalter @matt-royal

### CODE REVIEW

File: kpack-image-builder/controllers/imageprocessfetcher/image_process_fetcher.go [hunk 1]
```diff
@@ -66,8 +65,11 @@ func (f *ImageProcessFetcher) Fetch(imageRef string, credsOption remote.Option,
 
 // Reconstruct command with arguments into a single command string
 func extractFullCommand(process launch.Process) string {
-	commandWithArgs := append([]string{process.Command}, process.Args...)
-	return strings.Join(commandWithArgs, " ")
+	cmdString := process.Command
+	for _, a := range process.Args {
+		cmdString = fmt.Sprintf(`%s %q`, cmdString, a)
```
Comment (kieron-dev) [inline kpack-image-builder/controllers/imageprocessfetcher/image_process_fetcher.go:70]: I wonder if this is resilient to all arg content. Is it possible that a valid arg could contain quoting that would become invalid when wrapped in further quotes?

Comment (Birdrock) [reply kpack-image-builder/controllers/imageprocessfetcher/image_process_fetcher.go:70]: We should expect any valid args would be properly quoted and escaped. It is possible that manifest commands could come in without escaping, but there wouldn't be any args, so the quoting shouldn't come into play.

Regardless, this is more of a band-aid fix to enable buildpacks that provide a `bash -c` style cmd and args. We should avoid string processing of this sort whenever possible. We should have discussion around how to move forward.

Comment (davewalter) [reply kpack-image-builder/controllers/imageprocessfetcher/image_process_fetcher.go:70]: I think Kieron's concern is that, by wrapping a string with quotes without escaping the string being wrapped, something like `"properly \"escaped\" string"` becomes `""properly \"escaped\" string""`, which is not the same as `"\"properly \\\"escaped\\\" string\""`.

Comment (Birdrock) [reply kpack-image-builder/controllers/imageprocessfetcher/image_process_fetcher.go:70]: Looking at [fmt](https://pkg.go.dev/fmt) docs, `%q	a single-quoted character literal safely escaped with Go syntax.` I think we should be safe.

Comment (davewalter) [reply kpack-image-builder/controllers/imageprocessfetcher/image_process_fetcher.go:70]: I just tried this out in the Go playground and it looks to be doing what we want it to: https://go.dev/play/p/UZnII9jxCZe

**Review by davewalter — APPROVED** [review]

**Pull request merged**